twine/commands/upload.py

```diff
@@ -64,6 +64,11 @@ def upload(upload_settings: settings.Settings, dists: List[str]) -> None:
     repository = upload_settings.create_repository()
     uploaded_packages = []
 
+    if upload_settings.verbose:
+        print("\nList of packages that will be uploaded:")
+        for package in uploads:
```

Comment:
I think this is the source of the failing type checks (see the [Travis log](https://travis-ci.org/github/pypa/twine/jobs/695272199#L242-L251)), because this results in mypy inferring that `package` is a `str`, but later it's assigned a `PacakgeFile`  instance. You can resolve it by using a different variable name, e.g. `upload_filename`.